solr/solr-ref-guide/src/neural-search.adoc

```diff
@@ -0,0 +1,324 @@
+= Neural Search
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+Search comprises of performing four primary steps:
+
+* generate a representation of the query that specifies the information need
+* generate a representation of the document that captures the information contained
+* match the query and the document representations from the corpus of information
+* assign a score to each matched document in order to establish a meaningful document ranking by relevance in the results
+
+The Apache Solr *Neural Search* module adds support for neural networks based techniques that can improve various aspects of search.
+
+These techniques can be differentiated based on whether they affect the query representation, the document representation, or the estimation of the relevance score.
+
+Neural Search is an industry derivation from the academic field of https://www.microsoft.com/en-us/research/uploads/prod/2017/06/fntir2018-neuralir-mitra.pdf[Neural information Retrieval].
+
+== Neural Search Concepts
+
```

Comment:
This promises to discuss Neural Search Concepts, which IIUC includes a wide variety of topics and ideas. However, just like the name of the page, it's promising more than it delivers. There's mention of deep learning, but again we're not providing that - we're providing a way to index the _output_ of deep learning, which is then discussed. I think this section might need to be renamed and the following section on Deep Learning edited to be more focused on what Solr (via Lucene) *does* actually provide.

Reply:
The intention of the Neural Search concepts area is to give a nice and quick introduction to the user, including useful references.
If I rename it to "Important Concepts", any better?
Let me know, I would like to merge this contribution soon, if you have any additional text suggestions I can add them!